### PR TITLE
Optional hard-coded prefix for all tenant values, added with "-" delimiter.

### PR DESCRIPTION
**Pull request by melnikovio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,6 +107,9 @@ tenant:
   # regardless of the response from upstream. This can lose metrics if Cortex/Mimir is
   # throwing rejections.
   accept_all: false
+
+  # Optional hard-coded prefix for all tenant values, added with "-" delimiter.
+  tenant_prefix: foobar
```
Comment (blind-oracle) [inline README.md:112]: Let's maybe name it just `prefix`? Since we're anyway already in `tenant:` namespace.

File: config.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ type config struct {
 
 	Tenant struct {
 		Label       string
+		LabelPrefix string `yaml:"tenant_prefix"`
```
Comment (blind-oracle) [inline config.go:37]: Same here

File: processor.go [hunk 3]
```diff
@@ -313,6 +313,9 @@ func (p *processor) send(clientIP net.Addr, reqID uuid.UUID, tenant string, wr *
 	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
 	req.Header.Set("X-Cortex-Tenant-Client", clientIP.String())
 	req.Header.Set("X-Cortex-Tenant-ReqID", reqID.String())
+	if p.cfg.Tenant.LabelPrefix != "" {
+		tenant = fmt.Sprintf("%s-%s", p.cfg.Tenant.LabelPrefix, tenant)
```
Comment (blind-oracle) [inline processor.go:317]: I'm thinking maybe we should omit the `-` and let people choose themselves (maybe they want an underscore? :) )
So I'd just go with `tenant = p.cfg.Tenant.Prefix + tenant` (this should also be a bit faster than `fmt`)

Comment (WoozyMasta) [reply processor.go:317]: I would also provide a link or list the delimiters allowed for use in the README or comments to the config
https://grafana.com/docs/mimir/latest/configure/about-tenant-ids/

File: README.md (hunk 1 above)
Comment (melnikovio) [reply README.md]: Fixed. And I changed the code a bit to add some unit-tests for checking headers

File: processor.go (hunk 3 above)
Comment (melnikovio) [reply processor.go]: Fixed

**Comment by blind-oracle** [comment]
Thanks! One more thing - can you please run `gofmt` on all sources so that they're consistently formatted.

**Comment by melnikovio** [comment]
Sure! Sorry, completely forgot about it. Done

**Pull request merged**